scipy/stats/_distn_infrastructure.py

```diff
@@ -2302,6 +2305,8 @@ def fit(self, data, *args, **kwds):
             raise TypeError("Unknown arguments: %s." % kwds)
 
         vals = optimizer(func, x0, args=(ravel(data),), disp=0)
+        if _reduce_func is not None and get_tuple is not None:
+            vals = get_tuple(vals, ravel(data))
```

Comment:
See if we can replace `get_tuple` with re-defining `restore`

Reply:
implemented